### PR TITLE
net/tshttpproxy: fix incorrect type in Windows implementation, switch…

### PR DESCRIPTION
**Pull request by dblohm7**
… to mkwinsyscall, fix memory leak

The definition of winHTTPProxyInfo was using the wrong type (uint16 vs uint32)
for its first field. I fixed that type.

Furthermore, any UTF16 strings returned in that structure must be explicitly
freed. I added code to do this.

Finally, since this is the second time I've seen type safety errors in this code,
I switched the native API calls over to use wrappers generated by mkwinsyscall.
I know that would not have helped prevent the previous two problems, but every
bit helps IMHO.

Updates https://github.com/tailscale/tailscale/issues/4811

Signed-off-by: Aaron Klotz <aaron@tailscale.com>

### CODE REVIEW

File: net/tshttpproxy/tshttpproxy_windows.go [hunk 1]
```diff
@@ -221,30 +201,46 @@ type autoProxyOptions struct {
...
 // https://docs.microsoft.com/en-us/windows/win32/api/winhttp/ns-winhttp-winhttp_proxy_info
 type winHTTPProxyInfo struct {
-	AccessType  uint16
+	AccessType  uint32
 	Proxy       *uint16
 	ProxyBypass *uint16
 }
 
-var proxyForURLOpts = &autoProxyOptions{
+type winHGlobal windows.Handle
```
Comment (bradfitz) [inline net/tshttpproxy/tshttpproxy_windows.go:209]: why do we need this type?

File: net/tshttpproxy/mksyscall.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package tshttpproxy
+
+//go:generate go run golang.org/x/sys/windows/mkwinsyscall -output zsyscall_windows.go mksyscall.go
+
+//sys globalFree(hglobal winHGlobal) (err error) [failretval==0] = kernel32.GlobalFree
```
Comment (bradfitz) [inline net/tshttpproxy/mksyscall.go:5]: ugh I guess https://go-review.googlesource.com/c/sys/+/413435 stalled out

**Review by bradfitz — APPROVED** [review]
LGTM after license header fix

File: net/tshttpproxy/tshttpproxy_windows.go (hunk 1 above)
Comment (dblohm7) [reply net/tshttpproxy/tshttpproxy_windows.go:209]: In terms of "will it compile," we don't, but `HGLOBAL` is considered to be distinct from `HANDLE` as far as the SDK is concerned. I wanted to reflect that.

**Pull request merged**